docs/source/specification.rst

```diff
@@ -0,0 +1,20 @@
+.. _specification:
+
+====================================================
+The Reproducible Execution Environment Specification
+====================================================
+
+repo2docker scans a repository for particular :ref:`config_files`, such
+as ``requirements.txt`` or ``REQUIRE``. The collection of files and their contents
```

Comment:
```suggestion
as ``requirements.txt`` or ``REQUIRE``. The collection of files, their contents, and actions to take based on them
```
I think we should include something that says that the spec not only specifies which files are recognised, what their format/contents is but also specifies what you should do with each file.

In my mind this is what will form part of this spec for the `requirements.txt`:

> REE Spec recognises a `requirements.txt` as a valid config file. The file format is as defined by the `requirements.txt` standard of the Python community. A REE spec compliant tool will install a Python interpreter (of unspecified version) and perform the equivalent action of `pip install -r requirements.txt` so that the user can afterwards run `python` and use the packages installed.

The wording is not very good but in terms of topics covered this is what I imagine the spec will be.